tests/core/stream_muxer/test_yamux.py

```diff
@@ -0,0 +1,195 @@
+    client_yamux, server_yamux = yamux_pair
+    assert client_yamux.is_initiator
+    assert not server_yamux.is_initiator
+    with trio.move_on_after(5):
+        stream = await client_yamux.open_stream()
+        print("Stream opened")
+        assert isinstance(stream, YamuxStream)
+        assert stream.stream_id % 2 == 1
+    print("test_yamux_stream_creation complete")
+
+
+@pytest.mark.trio
+async def test_yamux_accept_stream(yamux_pair):
+    print("Starting test_yamux_accept_stream")
+    client_yamux, server_yamux = yamux_pair
+    client_stream = await client_yamux.open_stream()
+    server_stream = await server_yamux.accept_stream()
+    assert server_stream.stream_id == client_stream.stream_id
+    assert isinstance(server_stream, YamuxStream)
+    print("test_yamux_accept_stream complete")
+
+
+@pytest.mark.trio
+async def test_yamux_data_transfer(yamux_pair):
+    print("Starting test_yamux_data_transfer")
+    client_yamux, server_yamux = yamux_pair
+    client_stream = await client_yamux.open_stream()
+    server_stream = await server_yamux.accept_stream()
+    test_data = b"hello yamux"
+    await client_stream.write(test_data)
+    received = await server_stream.read(len(test_data))
+    assert received == test_data
+    reply_data = b"hi back"
+    await server_stream.write(reply_data)
+    received = await client_stream.read(len(reply_data))
+    assert received == reply_data
+    print("test_yamux_data_transfer complete")
+
+
+@pytest.mark.trio
+async def test_yamux_stream_close(yamux_pair):
+    print("Starting test_yamux_stream_close")
+    client_yamux, server_yamux = yamux_pair
+    client_stream = await client_yamux.open_stream()
+    server_stream = await server_yamux.accept_stream()
+    await client_stream.close()
+    received = await server_stream.read()
+    assert received == b""
+    assert client_stream.closed
+    with pytest.raises(MuxedStreamError):
+        await client_stream.write(b"test")
+    print("test_yamux_stream_close complete")
+
+
+@pytest.mark.trio
+async def test_yamux_stream_reset(yamux_pair):
+    print("Starting test_yamux_stream_reset")
+    client_yamux, server_yamux = yamux_pair
+    client_stream = await client_yamux.open_stream()
+    server_stream = await server_yamux.accept_stream()
+    await client_stream.reset()
+    data = await server_stream.read()
+    assert data == b"", "Expected empty read after reset"
+    print("test_yamux_stream_reset complete")
+
+
+@pytest.mark.trio
+async def test_yamux_connection_close(yamux_pair):
+    print("Starting test_yamux_connection_close")
+    client_yamux, server_yamux = yamux_pair
+    await client_yamux.open_stream()
+    await server_yamux.accept_stream()
+    await client_yamux.close()
+    print("Closing stream")
+    await trio.sleep(0.2)
+    assert client_yamux.is_closed
+    assert server_yamux.event_shutting_down.is_set()
+    print("test_yamux_connection_close complete")
+
+
+if __name__ == "__main__":
```

Comment:
I don't think we need this in a test file.

Reply:
Agreed. I’ve removed it from `test_yamux.py`. Thanks for pointing that out.